mantidproject/mantid
# Add data target for doctest builds

### Description of work
Explicitly build the `StandardTestData` target if `ENABLE_DOC_TESTS` is set to true. This is required now that `check_for_changes` [is working again](https://github.com/mantidproject/mantid/pull/37099). If a PR changes only docs, the flags `ENABLE_BUILD_CODE`, `ENABLE_UNIT_TESTS`, `ENABLE_SYSTEM_TESTS` are set to false to save CI resources. However, the doc tests would fail because the unit tests data wasn't being downloaded.

I've also had to skip the doctests in the `PeakMatching-v1.rst` because there is a `mantidqtinterfaces` dependency that causes an import to fail. These tests will still run in the nightly and any PR that doesn't only edit the docs. I've also created an [issue](https://github.com/mantidproject/mantid/issues/37145) to fix the underlying problem.

Additionally, a flag (`--disable-unittests`) has been added to explicitly enable the skipping of unit tests if required. This will be needed for moving doctests into their own PR check.



*There is no associated issue.*


### To test:
A test PR was set up to run only the doc tests along with the changes present in this PR. Check that it successfully runs the doctests even though the code build is skipped:
https://github.com/mantidproject/mantid/pull/37144/


*This does not require release notes* because **it's a fix to doctest build configuration**




---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
